mix.exs

```diff
@@ -16,21 +16,25 @@ defmodule EctoStateMachine.Mixfile do
    ]
   end
 
-  defp elixirc_paths(:prod), do: ["lib"]
-  defp elixirc_paths(:dev),  do: ["lib", "test/dummy/web", "test/dummy/lib"]
-  defp elixirc_paths(:test), do: ["lib", "test"]
+  defp elixirc_paths(:test), do: elixirc_paths ++ ["test/support"]
+  defp elixirc_paths(_),     do: elixirc_paths
+  defp elixirc_paths,        do: ["lib"]
 
   def application do
     [applications: [:logger]]
   end
 
+  def app_list(:test), do: app_list ++ [:ecto, :postgrex, :ex_machina]
+  def app_list(_),     do: app_list
+  def app_list,        do: [:logger]
+
   defp deps do
     [
      {:ecto, ">= 1.1.2 or >= 2.0.0-beta0"},
 
      {:postgrex,   ">= 0.0.0", only: :test},
-     {:ex_machina, "~> 1.0.0-beta.1", github: "thoughtbot/ex_machina", only: :test},
-     {:ex_spec,    "~> 1.0.0", only: :test}
+     {:ex_machina, "~> 1.0", only: :test},
```

Comment:
I tried to play this story fast. The problem is pretty simple: ex_machina from hex.pm will conflict with ecto version, since `ex_machina` 1.0 requires `ecto` 2.0. You brought 1.0 support.